carma-messenger-core/cpp_message/src/SPAT_Message.cpp

```diff
@@ -0,0 +1,702 @@
+                            if (state->states.list.array[j]->state_time_speed.list.array[k]->timing->likelyTime)
+                            {
+                                timing.likely_time_exists = true;
+                                DSRC_TimeMark_t *likely_time = new DSRC_TimeMark_t;
+                                likely_time = state->states.list.array[j]->state_time_speed.list.array[k]->timing->likelyTime;
+                                timing.likely_time = *likely_time;
+                            }
+
+                            timing.confidence_exists = false;
+                            if (state->states.list.array[j]->state_time_speed.list.array[k]->timing->confidence)
+                            {
+                                timing.confidence_exists = true;
+                                TimeIntervalConfidence_t *confidence = new TimeIntervalConfidence_t;
+                                confidence = state->states.list.array[j]->state_time_speed.list.array[k]->timing->confidence;
+                                timing.confidence = *confidence;
+                            }
+
+                            timing.next_time_exists = false;
+                            if (state->states.list.array[j]->state_time_speed.list.array[k]->timing->nextTime)
+                            {
+                                timing.next_time_exists = true;
+                                DSRC_TimeMark_t *next_time = new DSRC_TimeMark_t;
+                                next_time = state->states.list.array[j]->state_time_speed.list.array[k]->timing->nextTime;
+                                timing.next_time = *next_time;
+                            }
+                            movement_event.timing = timing;
+                        }
+                        //3. Advisory Speed List
+                        movement_event.speeds_exists = false;
+                        if (state->states.list.array[j]->state_time_speed.list.array[k]->speeds)
+                        {
+                            movement_event.speeds_exists = true;
+                            j2735_msgs::AdvisorySpeedList advisory_speed_list;
+                            for (size_t l = 0; l < state->states.list.array[j]->state_time_speed.list.array[k]->speeds->list.count; l++)
+                            {
+                                j2735_msgs::AdvisorySpeed advisory_speed;
+
+                                if (state->states.list.array[j]->state_time_speed.list.array[k]->speeds->list.array[l]->type)
+                                {
+                                    advisory_speed.type.advisory_speed_type = state->states.list.array[j]->state_time_speed.list.array[k]->speeds->list.array[l]->type;
+                                }
+                                else
+                                {
+                                    advisory_speed.type.advisory_speed_type = ADVISORY_SPEED_TYPE_NONE_;
+                                }
```

Comment:
advisory speed type is not optional it seems.
https://github.com/usdot-fhwa-stol/carma-msgs/blob/develop/j2735_msgs/msg/AdvisorySpeed.msg
and advisory_speed_type is also in bits, so ==1 check is needed I think

Reply:
By not optional do you mean there should be no if condition checking if a value exists?

Advisory speed type is also a single bit value from an enum https://github.com/usdot-fhwa-stol/carma-messenger/blob/develop/carma-messenger-core/cpp_message/include/generated/AdvisorySpeedType.h

Reply:
yeah I don't think we need to check if value exists